### PR TITLE
[Bug][Core]Check Read config file exist

### LINKED ISSUE #3484: [Bug] [Core] Read config file but without check it exist.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

image

- When we parse the config file, we always think it exists, which may throw `No configuration setting found for key 'env'` when the file does not exist. Users may be confused, we need to check whether it exists before parsing the config file, and if it does not exist, we need to throw an error and provide the correct prompt.
- Both Flink/Spark/ST Engine should do this check.

### SeaTunnel Version

dev

### SeaTunnel Config

```conf
-
```

### Running Command

```shell
-
```

### Error Exception

```log
-
```

### Flink or Spark Version

_No response_

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by TaoZex**
## Purpose of this pull request

https://github.com/apache/incubator-seatunnel/issues/3484
Check Read config file exist

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

File: seatunnel-core/seatunnel-flink-starter/src/main/java/org/apache/seatunnel/core/starter/flink/command/FlinkApiTaskExecuteCommand.java [hunk 1]
```diff
@@ -46,7 +49,14 @@ public FlinkApiTaskExecuteCommand(FlinkCommandArgs flinkCommandArgs) {
...
     public void execute() throws CommandExecuteException {
         Path configFile = FileUtils.getConfigPath(flinkCommandArgs);
-
+        if (!configFile.toFile().exists()) {
+            try {
+                throw new FileNotFoundException("Can't find config file: " + configFile);
+            } catch (FileNotFoundException e) {
+                log.error(ExceptionUtils.getMessage(e));
+                throw new SeaTunnelException(e);
+            }
```
Comment (hailin0) [inline seatunnel-core/seatunnel-flink-starter/src/main/java/org/apache/seatunnel/core/starter/flink/command/FlinkApiTaskExecuteCommand.java]:
```suggestion
            String message = "Can't find config file: " + configFile;
            log.error(message);
            throw new SeaTunnelException(message);
```

**Comment by Hisoka-X** [comment]
Also should do this check in Spark and ST engine.

File: seatunnel-core/seatunnel-flink-starter/src/main/java/org/apache/seatunnel/core/starter/flink/command/FlinkApiTaskExecuteCommand.java [hunk 2]
```diff
@@ -46,7 +47,11 @@ public FlinkApiTaskExecuteCommand(FlinkCommandArgs flinkCommandArgs) {
     @Override
     public void execute() throws CommandExecuteException {
         Path configFile = FileUtils.getConfigPath(flinkCommandArgs);
-
+        if (!configFile.toFile().exists()) {
```
Comment (Hisoka-X) [inline seatunnel-core/seatunnel-flink-starter/src/main/java/org/apache/seatunnel/core/starter/flink/command/FlinkApiTaskExecuteCommand.java]: All three type engine both use same FileUtils.getConfigPath, maybe you can put check logic in `FileUtils.getConfigPath`, you know, for redunce duplicate code.

Comment (TaoZex) [reply seatunnel-core/seatunnel-flink-starter/src/main/java/org/apache/seatunnel/core/starter/flink/command/FlinkApiTaskExecuteCommand.java]: > All three type engine both use same FileUtils.getConfigPath, maybe you can put check logic in `FileUtils.getConfigPath`, you know, for redunce duplicate code.

Thank you for your suggestion, it is a good way.

Comment (TaoZex) [reply seatunnel-core/seatunnel-flink-starter/src/main/java/org/apache/seatunnel/core/starter/flink/command/FlinkApiTaskExecuteCommand.java]: I have fixed it.

**Review by Hisoka-X — DISMISSED** [review]

**Comment by TaoZex** [comment]
@EricJoy2048 @hailin0 PTAL

File: seatunnel-core/seatunnel-core-starter/src/main/java/org/apache/seatunnel/core/starter/utils/FileUtils.java [hunk 3]
```diff
@@ -50,6 +54,19 @@ public static Path getConfigPath(AbstractCommandArgs args) {
...
 
+    /**
+     * Check whether the conf file exists.
+     *
+     * @param configFile the path of the config file
+     */
+    public static void checkConfigExist(Path configFile) {
+        if (!configFile.toFile().exists()) {
+            String message = "Can't find config file: " + configFile;
+            log.error(message);
```
Comment (TyrantLucifer) [inline seatunnel-core/seatunnel-core-starter/src/main/java/org/apache/seatunnel/core/starter/utils/FileUtils.java]: Do not `log.error` before throwing exception.

File: seatunnel-core/seatunnel-core-starter/src/main/java/org/apache/seatunnel/core/starter/utils/FileUtils.java [hunk 4]
```diff
@@ -50,6 +54,19 @@ public static Path getConfigPath(AbstractCommandArgs args) {
...
+            throw new SeaTunnelException(message);
```
Comment (TyrantLucifer) [inline seatunnel-core/seatunnel-core-starter/src/main/java/org/apache/seatunnel/core/starter/utils/FileUtils.java]: `SeaTunnelRuntimeException(CommonErrorCode.FILE_OPERATION_FAILED, message)` is better.

Comment (TaoZex) [reply seatunnel-core/seatunnel-core-starter/src/main/java/org/apache/seatunnel/core/starter/utils/FileUtils.java]: Thanks for you suggestion.I have fixed it.

**Review by TyrantLucifer — CHANGES_REQUESTED** [review]
Please change codes. Thanks.

**Comment by TaoZex** [comment]
> Please change codes. Thanks.

done.

**Review by Hisoka-X — APPROVED** [review]

**Review by TyrantLucifer — APPROVED** [review]

**Pull request merged**